### PR TITLE
Support multiple label columns

### PR DESCRIPTION
**Pull request by edknv**
Dataloader currently only returns the first label column if there are multiple **label** columns ([permalink](https://github.com/NVIDIA-Merlin/dataloader/blob/498543d54629216dc09a5854dca4dbeeabcab356/merlin/dataloader/loader_base.py#L626-L631)), but this might be problematic for Merlin Models and Transformers4rec, which expect a dictionary output for the labels when there are multiple labels. This PR implements support for when there are multiple label columns.

Note that Keras expects the name of the output layer to match and users will have to specify the output layer name if there multiple columns. This should not be an issue for Merlin Models.

#### Testing
Additional unit test.
Tested with https://github.com/NVIDIA-Merlin/models/pull/845 (manually with corresponding changes)

### CODE REVIEW

File: merlin/dataloader/loader_base.py [hunk 1]
```diff
@@ -626,9 +626,14 @@ def _handle_tensors(self, tensors, tensor_names):
         # TODO: use dict for labels as well?
```
Comment (oliverholworthy) [inline merlin/dataloader/loader_base.py:626]: I think this comment can be updated along with this this change. We can probably remove the TODO and replace with a description of what is going on here.

File: merlin/dataloader/loader_base.py [hunk 2]
```diff
@@ -626,9 +626,14 @@ def _handle_tensors(self, tensors, tensor_names):
...
         # would require output layers to match naming
         # labels should not exist separately they should be a regular column
-        labels = None
-        if len(self.label_names) > 0:
```
Comment (oliverholworthy) [inline merlin/dataloader/loader_base.py:630]: So this looks like it was previously not working for multiple columns. (we're silently ignoring targets and keeping them in the input data?)

Do we have any tests or functionality in Transformer4Rec that makes use of multiple targets? (cc. @sararb ) And if so, I'm wondering why this didn't cause any test to fail in this PR https://github.com/NVIDIA-Merlin/Transformers4Rec/pull/547 

I'd like to check that if we have muliple targets in Transformers4Rec and in PyTorch Models in general, that having the output of the 'y' here as a dict is going to work and the preferred option. (Which might inform whether or not we should make this output type configurable at this  stage.)

Comment (oliverholworthy) [reply merlin/dataloader/loader_base.py:630]: thanks for the explanation @sararb . So this PR looks like it will satisfy both Merlin Models and Transformers4Rec then with the three possible output types as you described in the last comment.

**Review by oliverholworthy — APPROVED** [review]
